### PR TITLE
Mark Activity and it's properties as Serializable 

### LINKED ISSUE #20640: Serializable attribute was removed on the System.Diagnostics.Activity
Activity was serializable because the current Activity is stored in the logical call context on .NET 4.5.

It was accidentally removed in https://github.com/dotnet/corefx/pull/19742 that causes SerizalizationException on .NET45 when Activity leaves AppDomain.

It only affects .NET45 and we do not have automated testing for it in corefx. We have tested it manually and agreed on the risk that it may break (#17262 (comment))

Our partners (ApplicationInsights) that consume this API do not have scenarios that involve cross AppDomain calls when the serialization breaks. So it was not detected there either

### PR DESCRIPTION
**Pull request by lmolkova**
Fixes #20640

Manually tested on .net 4.5

@danmosemsft  @krwq @stephentoub

### CODE REVIEW

**Comment by danmosemsft** [comment]
When this is tested and merged you'll want to make another PR, against release/2.0.0 branch.

**Comment by lmolkova** [comment]
> When this is tested and merged you'll want to make another PR, against release/2.0.0 branch.
sure

File: src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.Current.net45.cs [hunk 1]
```diff
@@ -32,8 +34,9 @@ private set
             }
         }
 
-#region private
-        
+        #region private
```
Comment (krwq) [inline src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.Current.net45.cs]: nit: whitespaces should be collapsed

**Comment by lmolkova** [comment]
System.Data.SqlClient.Tests.DiagnosticTest/ExecuteScalarAsyncTest test failed which I assume tracked by #20441

**Pull request merged**